src/MongoDB.Driver.Core/Core/Compression/Snappy/Native/SnappyNativeMethods.cs

```diff
@@ -0,0 +1,96 @@
+﻿/* Copyright 2019–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using MongoDB.Driver.Core.Compression.Native;
+using MongoDB.Driver.Core.NativeLibraryLoader;
+
+namespace MongoDB.Driver.Core.Compression.Snappy.Native
+{
+    internal static class SnappyNativeMethods
+    {
+        private static readonly IDictionary<SupportedPlatforms, string> __libraryPaths = new Dictionary<SupportedPlatforms, string>
+        {
+            { SupportedPlatforms.Windows, string.Empty }, // bin folder
+            // On Linux, the snappy library depends on the Snappy package.
+            // For Debian/Ubuntu: apt-get install libsnappy1 libsnappy-dev
+            { SupportedPlatforms.Linux, "../../native/linux/" }, // todo: check
+            { SupportedPlatforms.MacOS, "../../native/osx/" } // todo: check
+        };
+
+        private static readonly Delegates64.snappy_compress __snappy_compress;
+        private static readonly Delegates64.snappy_max_compressed_length __snappy_max_compressed_length;
+        private static readonly Delegates64.snappy_uncompress __snappy_uncompress;
+        private static readonly Delegates64.snappy_uncompressed_length __snappy_uncompressed_length;
+        private static readonly Delegates64.snappy_validate_compressed_buffer __snappy_validate_compressed_buffer;
```

Comment:
These should be Lazy to postpone any exceptions thrown until the first attempt (if ever) is made to call a native Snappy method.